core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -178,6 +178,40 @@ public void applyTo(TableMetadata.Builder metadataBuilder) {
     }
   }
 
+  class SetStatistics implements MetadataUpdate {
```

Comment:
Can you add support for these to `MetadataUpdateParser`?

Reply:
Added changes in `MetadataUpdateParser` along with tests in `TestMetadataUpdateParser`, please take a look!